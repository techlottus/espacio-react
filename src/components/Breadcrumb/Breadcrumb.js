import React, { useEffect } from "react";

export const Breadcrumb = React.memo(({ data, onItem, onBack }) => {
  const breadcrumbRef = React.createRef();

  useEffect(() => {
    breadcrumbRef.current.data = {
      textItems: data.textItems || [],
      textColor: data.textColor || "brand",
      // iconColor: data.iconColor || '',
      // itemFocusColor: data.itemFocusColor || '',
      icon: data.icon || "",
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    breadcrumbRef.current.addEventListener("onItem", onItem);
    breadcrumbRef.current.addEventListener("onBack", onBack);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-breadcrumb ref={breadcrumbRef}></lottus-breadcrumb>
    </>
  );
});
