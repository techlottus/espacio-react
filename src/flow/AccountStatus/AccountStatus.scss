@import '../../App.scss';

.accountstatus-container {
  position: relative;
  width: 100%;
  font-family: 'Nunito', sans-serif;
  display: flex;
  flex-direction: column;
  background-color: map-get($palette-neutral, n05);
  color: map-get($palette-neutral, n80);
  @extend .headerapp-top;
  @extend .footer-bottom;


  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    width: 46%;
    padding: 0 27%;
    background-color: transparent;
  }

  > .accountstatusbackground {
    position: absolute;
    z-index: -1;
    left: 0;
    width: 100%;
    height: pixeltorem(277px);
    background-color: map-get($palette-neutral, n05);
    display: none;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      display: block;
    }
  }

  > .accountstatusbreadcrumb {
    width: calc(100% - #{pixeltorem(32px)});
    margin-top: pixeltorem(36px);
    padding: 0 pixeltorem(16px);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      width: 100%;
      padding: 0;
    }
  }

  .accountstatustitle {
    width: calc(100% - #{pixeltorem(32px)});
    margin-top: pixeltorem(40px);
    font-size: pixeltorem(24px);
    line-height: pixeltorem(30px);
    font-family: 'Nunito Bold',sans-serif;
    padding: 0 pixeltorem(16px);
    margin-bottom: pixeltorem(20px);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      width: 100%;
      padding: 0;
    }
  }

  > .accountstatuscontent {
    box-shadow: 0 pixeltorem(20px) pixeltorem(60px) rgba(0, 0, 0, 0.15);
    width: 100%;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      margin-bottom: pixeltorem(48px);
      border-radius: pixeltorem(8px);
    }
  }
}

.feedbackAccount {
  @extend .alertfeedbackcmp; 

}
