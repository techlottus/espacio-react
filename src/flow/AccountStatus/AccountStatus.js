import React, { useEffect } from "react";
import "./AccountStatus.scss";
import { Breadcrumb } from "../../components/Breadcrumb/Breadcrumb";
import HeaderApp from "../../shared/HeaderApp/HeaderApp";
import {
  breadcrumb,
} from "../../constants/AccountStatus.constant";
import AccountStatusResume from "./components/AccountStatusResume.js/AccountStatusResume";
import FooterApp from "../../shared/FooterApp/FooterApp";
import { useDispatch, useSelector } from "react-redux";
import { getAllAccountsService } from "../../middlewares/accountStatusMiddleware";
import { useHistory } from "react-router";
import AccountStatusPeriodDetails from "./components/AccountPeriodDetails/AccountStatusPeriodDetails";
import { useContainerHeigh } from "../../hooks/contentHeight";

const AccountStatus = () => {
  const {height} = useContainerHeigh(0)

  const history = useHistory();
  const dispatch = useDispatch();
  const { texts } = useSelector((state) => state.texts);

  useEffect(() => {
    window.scrollTo(0, 0);
    dispatch(getAllAccountsService());
  }, []); // eslint-disable-line react-hooks/exhaustive-deps
  

  const onItem = (e) => {
    if (e.detail === "/") {
      history.push(e.detail);
    }
  };

  const onBack = () => {
    history.push("/");
  };

  return (
    <>
      <HeaderApp />
      <div className="accountstatus-container" style = {{minHeight: `calc(${height}px - 77px)`}}>
        <div className="accountstatusbackground"></div>
        <div className="accountstatusbreadcrumb">
          <Breadcrumb data={breadcrumb} onItem={onItem} onBack={onBack} />
        </div>
        <div className="accountstatustitle">
          {texts?.accountStatus?.statusTitle}
        </div>
        <div className="accountstatuscontent">
          <AccountStatusResume />
        </div>
        <div>
          <AccountStatusPeriodDetails />
        </div>
      </div>
      <FooterApp />
    </>
  );
};

export default AccountStatus;
