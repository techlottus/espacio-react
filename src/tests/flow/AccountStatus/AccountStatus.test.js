import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";
import AccountStatus from "../../../flow/AccountStatus/AccountStatus";
import { textDashboardMock } from "../../../fixtures/dashboardFixture";
import { textsAccountStatusMock } from "../../../fixtures/accountStatusFixture";
import { typesRequestErrors } from "../../../constants/error.constant";
import { act } from "react-dom/test-utils";
import { Breadcrumb } from "../../../components/Breadcrumb/Breadcrumb";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();

let initState = {
  accountStatus: {
    profile: null,
    movements: null,
    errors: {
      [typesRequestErrors.getAccountStatus]: {
        isError: false,
        msg: null,
      },
      [typesRequestErrors.getAccountMovements]: {
        isError: false,
        msg: null,
      },
    },
    programs: null,
    success: {
      isSuccess: false,
      msg: null,
    },
    debitPeriods: null,
  },
  texts: {
    accountStatus: {
      ...textsAccountStatusMock
    },
    dashboard: {
      ...textDashboardMock
    }
  }
};

const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <BrowserRouter>
        <AccountStatus history={historyMock} />
      </BrowserRouter>
    </Provider>
  );
};

describe("Test <AccountStatus/>", () => {
  beforeEach(() => {
    store = mockStore(initState);
  });

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  describe('should run action',() => {
    test('on item from home',() => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Breadcrumb).props().onItem({
          detail: '/'
        });
      })
      wrapper.update();
    })

    test('on item diferrent from home',() => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Breadcrumb).props().onItem({
          detail: '/payment'
        });
      })
      wrapper.update();
    })


    test('on back',() => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Breadcrumb).props().onBack();
      })
      wrapper.update();
    })
  })

 

  
});
