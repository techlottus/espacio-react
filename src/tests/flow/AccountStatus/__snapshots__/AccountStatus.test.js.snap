// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AccountStatus/> should init component 1`] = `
Array [
  <div
    className="navbar"
  >
    <lottus-navbar />
  </div>,
  <div
    className="accountstatus-container"
  >
    <div
      className="accountstatusbackground"
    />
    <div
      className="accountstatusbreadcrumb"
    >
      <lottus-breadcrumb />
    </div>
    <div
      className="accountstatustitle"
    />
    <div
      className="accountstatuscontent"
    >
      <div
        className="accountstatusresume-container"
      >
        <div
          className="accountstatuserror"
        />
        <div
          className="accountstatusresumeperfil"
        />
        <div
          className="accountstatusresumebtn"
        >
          <div
            className="downloadbtn"
          >
            <lottus-button />
          </div>
          <lottus-button />
        </div>
      </div>
    </div>
    <div>
      <div
        className="accountdetails-container"
      >
        <h2
          className="headingdetails"
        >
          Detalles por periodo
        </h2>
        <p
          className="textdetails"
        >
          Selecciona el programa y el periodo para ver los detalles de los movimientos; se clasifican de la siguiente manera:
        </p>
        <div
          className="detail-bullets"
        >
          <p
            className="org-p"
          >
            <span
              className="org-bullet"
            />
            Con adeudo
          </p>
          <p
            className="gr-p"
          >
            <span
              className="gr-bullet"
            />
            Pagados
          </p>
          <p
            className="pr-p"
          >
            <span
              className="pr-bullet"
            />
            Por pagar
          </p>
        </div>
        <div
          className="accountdetails-selects"
        >
          <div
            className="select-details"
          >
            <p
              className="textprogram"
            >
              Programa
            </p>
            <lottus-select />
          </div>
          <div
            className="select-details"
          >
            <p
              className="textprogram"
            >
              Periodo
            </p>
            <lottus-select />
          </div>
        </div>
        <div
          className="accountdetails-period"
        >
          <div>
            <h2
              className="headingdetails"
            >
              Periodo 
            </h2>
            <div
              className="calendar"
            >
              <span
                className="material-icons icon iconright"
              >
                calendar_today
              </span>
              <p
                className="textperiod"
              />
            </div>
          </div>
          <div>
            <p>
              Saldo del periodo
            </p>
            <p
              className="detail-amount textperiod"
            >
              $0.00
            </p>
          </div>
        </div>
        <div>
          <lottus-label />
        </div>
        <div
          className="accountdetails-movements"
        >
          <h2
            className="headingdetails"
          >
            Movimientos
          </h2>
          <div />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="footer"
    style={
      Object {
        "bottom": "0",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <lottus-footer />
  </div>,
]
`;
